### PR TITLE
expose the parseMetadata function and all required classes to use it

### PR DESCRIPTION
**Pull request by jakemac53**
Towards https://github.com/dart-lang/build/issues/1554 (makes that a trivial change in build_test)

### CODE REVIEW

File: pkgs/test_core/lib/src/runner/parse_metadata.dart [hunk 1]
```diff
@@ -24,8 +24,11 @@ import '../util/dart.dart';
 ///
 /// Throws an [AnalysisError] if parsing fails or a [FormatException] if the
 /// test annotations are incorrect.
-Metadata parseMetadata(String path, Set<String> platformVariables) =>
-    _Parser(path, platformVariables).parse();
+///
+/// If [contents] is `null` then it will be read as a [File] using [path].
+Metadata parseMetadata(String path, Set<String> platformVariables,
```
Comment (natebosch) [inline pkgs/test_core/lib/src/runner/parse_metadata.dart]: I would make this always take `content` and update the calling locations to read the file themselves.

File: pkgs/test_core/lib/backend.dart [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+export 'package:test_api/src/backend/metadata.dart' show Metadata;
```
Comment (natebosch) [inline pkgs/test_core/lib/backend.dart]: should this be `export 'package:test_api/backend.dart' show ...;`?

File: pkgs/test_api/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 0.2.5
+
+* Expose the  `Metadata`, `PlatformSelector`, `Runtime`, and `SuitePlatform`
```
Comment (natebosch) [inline pkgs/test_api/CHANGELOG.md]: Do you want to take a pass through other code using this and see if we can get rid of any existing `src/` imports from the other packages? Or should we lump that in with the other cleanup later?

Comment (jakemac53) [reply pkgs/test_api/CHANGELOG.md]: I just did that for `test_core` for a few imports, but ya I think we can handle that for other packages later. We could probably search all the packages we pull internally the easiest?

File: pkgs/test_core/lib/backend.dart (hunk 2 above)
Comment (jakemac53) [reply pkgs/test_core/lib/backend.dart]: done

File: pkgs/test_core/lib/src/runner/parse_metadata.dart (hunk 1 above)
Comment (natebosch) [reply pkgs/test_core/lib/src/runner/parse_metadata.dart]: Did this slip?

Is it still worth doing? We haven't published yet...

Comment (jakemac53) [reply pkgs/test_core/lib/src/runner/parse_metadata.dart]: Oh weird - somehow I didn't see this comment when I was merging. I will go ahead and send out another PR for it.

**Pull request merged**